setup.py

```diff
@@ -78,7 +78,7 @@ def _dbt_core_version(plugin_version: str) -> str:
         "google-cloud-bigquery~=3.0",
         "google-cloud-storage~=2.4",
         "google-cloud-dataproc~=5.0",
-        "agate~=1.6.3",
+        "agate~=1.7.0",
```

Comment:
@colin-rogers-dbt Looks like you added this back in #458 (d27cf24e3bfadc0f1041bea36b6d7fd890e975a8)

Reply:
I can't remember why, if it's causing issues we can certainly remove it

Reply:
We should keep `agate` if we're using it directly, but without a pin.

We ran into an issue a few months ago where a requirement was dropped from `dbt-core` that was used directly in an adapter. To avoid version conflicts, we did not explicitly list it as a requirement in the adapter; hence the adapter broke when we removed it from `dbt-core`. This is likely a rare occurrence, but I think this is the right compromise with the least downsides. I did that here in `dbt-redshift`:

https://github.com/dbt-labs/dbt-redshift/blob/36bed56310645669a9bf20b882e6aed51bf1b98a/setup.py#L89-L90

Reply:
@mikealfare Heard! I removed it in https://github.com/dbt-labs/dbt-bigquery/pull/778; shall I open a new PR to add it back (without a pin)?